### PR TITLE
Skip rocm failing tests

### PR DESCRIPTION
**Pull request by pnunna93**
This PR skips all tests that fail on ROCm

### CODE REVIEW

File: tests/test_autograd.py [hunk 1]
```diff
@@ -4,6 +4,7 @@
 import torch
 
 import bitsandbytes as bnb
+from bitsandbytes.cextension import HIP_ENVIRONMENT
```
Comment (amathews-amd) [inline tests/test_autograd.py:7]: If this is run on NV, will this fail ?

**Review by jpvillam-amd — APPROVED** [review]
LGTM

File: tests/test_autograd.py (hunk 1 above)
Comment (pnunna93) [reply tests/test_autograd.py:7]: Yes, I think so, thanks for pointing it out @amathews-amd. I will add a default value.

Comment (pnunna93) [reply tests/test_autograd.py:7]: Fixed it here: https://github.com/ROCmSoftwarePlatform/bitsandbytes/pull/1/commits/28b80564b7efbec22eba9800ac4b83ee745a34c1

**Review by amathews-amd — APPROVED** [review]
LGTM

**Pull request merged**